src/Microsoft.AspNet.Mvc.Core/Rendering/Expressions/ViewDataEvaluator.cs

```diff
@@ -1,16 +1,23 @@
 // Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
+using System;
 using System.Collections.Generic;
 using System.Reflection;
+using Microsoft.AspNet.Mvc.Core;
 using Microsoft.Framework.Internal;
 
 namespace Microsoft.AspNet.Mvc.Rendering.Expressions
 {
     public static class ViewDataEvaluator
     {
-        public static ViewDataInfo Eval([NotNull] ViewDataDictionary viewData, [NotNull] string expression)
+        public static ViewDataInfo Eval([NotNull] ViewDataDictionary viewData, string expression)
         {
+            if (string.IsNullOrEmpty(expression))
```

Comment:
:-1: I'm just moving code around to avoid a bit of duplication.

in any case, whitespace would be disallowed at a higher level because it does not match a supported expression name.  however `null` and empty are special-cased throughout MVC.  just not supported way down here.